Chrysalis-Papilio/Soen-490-Papilio-Android
# Upcoming Activities Page [Papilio 28]

# General info

**Issue number**: PAPILIO-28

**Task description**: 

 - Added ability to view all the user's created and joined activities all on one page, the upcoming activities page

# Testing
To test this feature you have to:

1. Log in or sign up to the app
2. Create an activity. Note the date of this activity. 
3. Go to the "My Activities" tab and the first thing you'll see is the upcoming activities page. Wait for the upcoming activities to load. 
4. Once loading is done, you will see a chronological view of activities ordered by the date (from now to end of December). Look for the activity you created here. 
5. Verify that the activity you created appears on this page under the correct month. Also verify that in the activity card, this symbol is shown:

![image](https://user-images.githubusercontent.com/9057422/224882593-8e3a9ff8-ce81-4c42-9408-1d71fff2b98e.png)

6. Also verify that you can tap on the card and get taken to that activity's details page. 
7. Now go back to the Activity Feed page and join any activity that you have not created. 
8. Now go to the Upcoming Activities page again and wait for it to load again
9. Verify that the activity you recently joined is shown here under the correct month. Also verify that the activity card does NOT show the symbol above. 
10. Verify that tapping on the month container of the activity hides/shows activities for that month.  
11. If steps 5, 6, 9, and 10 are verified in the affirmative, then this story is complete. 

# Notes
- UI subject to improvements later on.
